aspnetcore/fundamentals/http-context.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Access HttpContext in ASP.NET Core
+author: coderandhiker
+description: Learn how to access HttpContext in ASP.NET Core.
+ms.author: riande
+ms.custom: mvc
+ms.date: 07/10/2018
+uid: fundamentals/httpcontext
+---
+# Access HttpContext in ASP.NET Core
+
+ASP.NET Core apps access the `HttpContext` through the [IHttpContextAccessor](/dotnet/api/microsoft.aspnetcore.http.ihttpcontextaccessor) interface and its default implementation [HttpContextAccessor](/dotnet/api/microsoft.aspnetcore.http.httpcontextaccessor).
+
+::: moniker range=">= aspnetcore-2.0"
+
+## Use HttpContext from Razor Pages
+
+The Razor Pages [PageModel](/dotnet/api/microsoft.aspnetcore.mvc.razorpages.pagemodel) exposes the [HttpContext](/dotnet/api/microsoft.aspnetcore.mvc.razorpages.pagemodel.httpcontext) property:
+
+```csharp
+public class AboutModel : PageModel
+{
+    public string Message { get; set; }
+
+    public void OnGet()
+    {
+        Message = HttpContext.Request.PathBase;
+    }
+}
+```
+
+::: moniker-end
+
+## Use HttpContext from a controller
+
+Controllers expose the `HttpContext` property from the [ControllerBase](/dotnet/api/microsoft.aspnetcore.mvc.controllerbase):
+
+```csharp
+public IActionResult About()
```

Comment:
I'm confused by this example. `ControllerBase` is for controllers not needing view support, but this action returns a view.

Reply:
It's still inherited via ControllerBase tho. ControllerBase -> Controller > HttpContext

Reply:
The confusion stems from the `ControllerBase.` part before the `HttpContext` property name. I'll update the code snippet to include the controller declaration inheriting from `Controller`. That will eliminate any confusion in the future. Some readers may think this means inherit from `ControllerBase`, which won't compile in this example.

Reply:
I see. :+1: